### PR TITLE
[stable/prometheus-operator] Add support of bearer token for Prometheus' ServiceMonitor

### PR DESCRIPTION
**Pull request by cryptobioz**
#### What this PR does / why we need it:

As we use [oauth-proxy](https://github.com/openshift/oauth-proxy) containers to protect our Prometheus servers, the `/metrics` cannot be scraped without authentication. Thus, the _self-scraping target_ appears to be down (the `oauth-proxy` returns a `403 Forbidden`).
To allow the Prometheus server to scrape its own metrics, it must use a bearer token to pass through the proxy.
This PR adds the support of bearer tokens in the Prometheus' ServiceMonitor.

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/prometheus-operator/templates/prometheus/servicemonitor.yaml [hunk 1]
```diff
@@ -21,6 +21,9 @@ spec:
     {{- if .Values.prometheus.serviceMonitor.interval }}
     interval: {{ .Values.prometheus.serviceMonitor.interval }}
     {{- end }}
+    {{- if .Values.prometheus.serviceMonitor.bearerTokenFile }}
+    bearerTokenFile: {{ .Values.prometheus.serviceMonitor.bearerTokenFile }}
```
Comment (vsliouniaev) [inline stable/prometheus-operator/templates/prometheus/servicemonitor.yaml:25]: This should appear in the values file as well

**Review by vsliouniaev — CHANGES_REQUESTED** [review]

File: stable/prometheus-operator/templates/prometheus/servicemonitor.yaml (hunk 1 above)
Comment (cryptobioz) [reply stable/prometheus-operator/templates/prometheus/servicemonitor.yaml:25]: The variable has been added to the values file.

**Comment by vsliouniaev** [comment]
Should get better with https://github.com/helm/charts/pull/18430

**Comment by vsliouniaev** [comment]
Will merge once conflicts are resolved!
/lgtm

**Pull request merged**